### PR TITLE
Add support for contextual multi-armed bandits

### PR DESCRIPTION
**Pull request by aarsilv**
_Eppo Internal:_
🎟️ **Ticket:** [FF-2405 - Add bandit to common JavaScript SDK](https://linear.app/eppo/issue/FF-2405/[parent]-add-bandit-to-common-javascript-sdk)
🗺️ **Design Review:** [Bandits Engineering Design](https://www.notion.so/eppo/Bandits-Engineering-Design-c5df5b6072b446aa8cacbf54e4d1f9df?pvs=4)

## Motivation and Context
We want to add bandit capabilities ([high-level documentation](https://docs.geteppo.com/sdks/sdk-features/bandits/)) to our Node SDK and, possibly, our Java Client and React Native SDKs in the future.

### Core Change
* The configuration requestor will also fetch and store bandit parameters if any bandits are at play
* `getBanditAction()` can be used to request an assigned `variation` and `action` tuple from a flag that involves a bandit. If the bandit is the assigned variation, `action` will have the bandit-selected action (otherwise it will be `null`).

### Incremental Pull Requests
* #85 
* #88 
* #95
* #104 
* #108

### Draft Upstream Pull Requests
* `js-client-sdk #92` - [Update for updated common SDK](https://github.com/Eppo-exp/js-client-sdk/pull/92)
* `node-server-sdk #64` - [Add support for contextual bandits](https://github.com/Eppo-exp/node-server-sdk/pull/64)

## How has this been tested?
* Installed with the JavaScript Client and Node SDKs and ensured correct operation
  * Client SDK continues to function as expected
  * Node SDK passes shared bandit tests 
* Updated existing tests, including:
  * `configuration-requestor.spec.ts`   
  * `eppo-client.spec.ts` 
* Added test file for the newly-created bandit evaluator, where most of the bandit logic is contained
  * `bandit-evaluator.spec.ts`
* Added a new test file for the client for the shared SDK bandit tests as well as client-specific bandit tests
  * `eppo-client-with-bandits.spec.ts`
  image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@eppo/js-client-sdk-common",
-  "version": "3.4.0",
+  "version": "3.5.0",
```
Comment (aarsilv) [inline package.json:3]: 📈

File: src/client/eppo-client.ts [hunk 2]
```diff
@@ -133,7 +172,17 @@ export interface IEppoClient {
...
+
+  setBanditVariationConfigurationStore(
+    banditVariationConfigurationStore: IConfigurationStore<BanditVariation[]>,
+  ): void;
+
+  setBanditModelConfigurationStore(
+    banditModelConfigurationStore: IConfigurationStore<BanditParameters>,
+  ): void;
+
+  setIsObfuscated(isObfuscated: boolean): void;
```
Comment (aarsilv) [inline src/client/eppo-client.ts:185]: ` setIsObfuscated()` wasn't added to the `IEppoClient` interface so figured I'd hoist it up along with the above newly-added bandit store setters 🧹

File: src/constants.ts [hunk 3]
```diff
@@ -6,7 +6,9 @@ export const DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES = 1;
 export const DEFAULT_POLL_CONFIG_REQUEST_RETRIES = 7;
 export const BASE_URL = 'https://fscdn.eppo.cloud/api';
 export const UFC_ENDPOINT = '/flag-config/v1/config';
+export const BANDIT_ENDPOINT = '/flag-config/v1/bandits';
 export const SESSION_ASSIGNMENT_CONFIG_LOADED = 'eppo-session-assignment-config-loaded';
 export const NULL_SENTINEL = 'EPPO_NULL';
 // number of logging events that may be queued while waiting for initialization
 export const MAX_EVENT_QUEUE_SIZE = 100;
+export const BANDIT_ASSIGNMENT_SHARDS = 10000;
```
Comment (giorgiomartini0) [inline src/constants.ts:14]: 👌

File: src/client/eppo-client.spec.ts [hunk 4]
```diff
@@ -104,19 +111,19 @@ describe('EppoClient E2E test', () => {
...
       client.setIsGracefulFailureMode(true);
 
-      expect(client.getBoolAssignment(flagKey, 'subject-identifer', {}, true)).toBe(true);
-      expect(client.getBoolAssignment(flagKey, 'subject-identifer', {}, false)).toBe(false);
-      expect(client.getBooleanAssignment(flagKey, 'subject-identifer', {}, true)).toBe(true);
-      expect(client.getBooleanAssignment(flagKey, 'subject-identifer', {}, false)).toBe(false);
-      expect(client.getNumericAssignment(flagKey, 'subject-identifer', {}, 1)).toBe(1);
-      expect(client.getNumericAssignment(flagKey, 'subject-identifer', {}, 0)).toBe(0);
-      expect(client.getJSONAssignment(flagKey, 'subject-identifer', {}, {})).toEqual({});
+      expect(client.getBoolAssignment(flagKey, 'subject-identifier', {}, true)).toBe(true);
```
Comment (giorgiomartini0) [inline src/client/eppo-client.spec.ts:114]: 🧹 ❤️

**Review by giorgiomartini0 — APPROVED** [review]
Only skimmed since most of this is familiar - everything looks good!

**Pull request merged**